task7/Working with posts.js

```diff
@@ -0,0 +1,353 @@
+	}
+
+	get liesInInterval() {
+		return this._liesInInterval;
+	}
+
+	_hasAnyTag(hashTagsOfPost, hashTags) {
+		return hashTagsOfPost.some(hashTagOfPost=>(hashTags.includes(hashTagOfPost)));
+	}
+
+	get hasAnyTag() {
+		return this._hasAnyTag;
+	}
+
+	_compDates(x, y) {
+		return x.createdAt - y.createdAt;
+	}
+
+	get compDates() {
+		return this._compDates;
+	}
+
+	_isTag(str) {
+		return str[0] == '#';
+	}
+
+	get isTag() {
+		return this._isTag;
+	}
+
+	_isString(obj) {
+		return typeof obj == 'string';
+	}
+
+	get isString() {
+		return this._isString;
+	}
+
+  _rightDate(date) {
+		return date.getHours() >= 0 && date.getHours() <= 23 &&
+		 date.getMinutes() >= 0 && date.getMinutes() <= 59 &&
+		 	 date.getSeconds() >= 0 && date.getSeconds() <= 59;
+	}
+
+	get rightDate() {
+		return this._rightDate;
+	}
+
+	_getPostsByFilter(filterConfig) {
+		return this.posts.filter(element=>
+			(filterConfig.author == "" || element.author == filterConfig.author) &&
+			((filterConfig.dateInterval == null) || this.liesInInterval(element.createdAt, filterConfig.dateInterval)) &&
+			 ((filterConfig.hashTags == []) || this.hasAnyTag(element.hashTags, filterConfig.hashTags))
+			 );
+	}
+
+	get getPostsByFilter() {
+		return this._getPostsByFilter;
+	}
+
+	getPosts(filterConfig, skip=0, top=10) {
+		if (skip < 0 || top <= 0) {
+			return [];
+		}
+		let postsByFilter = this.getPostsByFilter(filterConfig).sort(this.compDates);
+		return postsByFilter.slice(skip, skip + top);
+	}
+
+	getPost(ID) {
+		return this.posts.find(item=>item.id == ID);
+	}
+
+	validatePost(somePost) {
+		return this.isString(somePost.description)&&
+		 somePost.description.length > 0 && somePost.description.length < 200 &&
+		 this.isString(somePost.author) && somePost.author.length > 0 &&
+		 somePost.hashTags.every(this.isTag) &&
+		 somePost.hashTags.every(this.isString) && somePost.likes.every(this.isString) &&
+		 somePost.createdAt instanceof Date && this.rightDate(somePost.createdAt);
+	}
+
+	addPost(somePost) {
+		if (!this.validatePost(somePost)) {
+			return false;
+		}
+		this.posts.push(somePost);
+		this.count += 1;
+		return true;
+	}
+
+	editPost(ID, edit) {
+		let index = this.posts.findIndex(item=>item.id==ID);
+		if (index==-1) {
+			return false;
+		}
+		let edPost = this.posts[index];
+		if (edit.description != "") {
+			edPost.description = edit.description;
+		}
+		if (edit.hashTags.length != 0) {
+			edPost.hashTags = edit.hashTags;
+		}
+		if (edit.photoLink != "") {
+			edPost.photoLink = edit.photoLink;
+		}
+		if (!this.validatePost(edPost)) {
+			return false;
+		}
+		this.posts[index] = edPost;
+		return true;
+	}
+
+	removePost(ID) {
+		let index = this.posts.findIndex(item=>item.id==ID);
+		if (index == -1) {
+			return false;
+		}
+		this.posts.splice(index, 1);
+		this.count -= 1;
+		return true;
+	}
+
+	addAll(arrayPosts) {
+		let invalidPosts = [];
+		arrayPosts.forEach(item => {
```

Comment:
Can you try to use `reduce` method for it? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce